wadas/domain/motion_detection.py

```diff
@@ -0,0 +1,61 @@
+"""Module containing logic to instantiate a Motion Detection thread to process video stream
+   integrated with Qt UI."""
+
+import time
+
+import cv2
+from PySide6.QtCore import QThread, Signal
+from PySide6.QtGui import QImage
+
+from wadas.domain.camera import Camera
+
+
+class MotionDetectionThread(QThread):
+    """Motion Detection QThread to integrate cv video stream into Qt Dialog"""
+
+    frame_ready = Signal(QImage)
+
+    def __init__(self, camera_index=0, parent=None):
+        super().__init__(parent)
+        self.running = False
+        self.camera_index = camera_index
+
+    def run(self):
+        self.running = True
+        cap = cv2.VideoCapture(self.camera_index)
+        background_sub = cv2.createBackgroundSubtractorMOG2()
+
+        if not cap.isOpened():
+            print("Error: Unable to open camera.")
+            return
+
+        while self.running:
+            ret, frame = cap.read()
```

Comment:
What's the meaning of ret?

Reply:
It's a bool composing the returning tuple

Reply:
I checked the cv2 documentation. Basically ret signals if the .read() was successful or not. So, "success" might be a better identifier instead of ret.
Anyway, this API was very badly defined: it should have given back just a single return value instead of a tuple (either the image itself or None).